File: solutions/Length-of-Last-Word/Length-of-Last-Word.py


class Solution:
    # @param s, a string
    # @return an integer
    def lengthOfLastWord(self, s):
        if len(s)==0:
            return 0
        start=-1
        for i in range(len(s)-1,-1,-1):
            if s[i]!=' ':
                start=i;
                break;
        if start==-1:
            return 0
        end=-1
        for i in range(start-1,-1,-1):
            if s[i]==' ':
                end=i
                break
        return start-end
 

File: solutions/Length-of-Last-Word/Length-of-Last-Word.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function (s) {
  const n = s.length;
  let ans = 0;
  for (let i = 0; i < n; i++) {
    if (s[i] !== ' ') {
      ans++;
    } else if (i + 1 < n && s[i + 1] !== ' ') {
      ans = 0;
    }
  }
  return ans;
};